client/extensions/woocommerce/state/ui/products/selectors.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { get, find, isNumber } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getProduct } from '../../products/selectors';
+
+function getEditsState( state ) {
+	const woocommerce = state.extensions.woocommerce;
+	return get( woocommerce, 'ui.products.edits' );
+}
+
+/**
+ * Gets the accumulated edits for a product, if any.
+ *
+ * @param {Object} state Global state tree
+ * @param {any} productId The id of the product (or { index: # } )
+ * @return {Object} The current accumulated edits
+ */
+export function getProductEdits( state, productId ) {
+	const edits = getEditsState( state ) || {};
+	const bucket = isNumber( productId ) && 'updates' || 'creates';
+	const array = edits[ bucket ] || [];
+
+	return find( array, ( p ) => productId === p.id );
+}
+
+/**
+ * Gets a combined product ( fetched data + edits )
+ *
+ * @param {Object} state Global state tree
+ * @param {any} productId The id of the product (or { index: # } )
+ * @return {Object} The product data merged between the fetched data and edits
+ */
+export function getCombinedProduct( state, productId ) {
+	const existing = isNumber( productId );
+
+	const product = existing && getProduct( state, productId );
+	const productEdits = getProductEdits( state, productId );
+
+	return ( product || productEdits ) && { ...product, ...productEdits } || undefined;
+}
+
+/**
+ * Gets the product being currently edited in the UI.
+ *
+ * @param {Object} state Global state tree
+ * @return {Object} Product object that is merged between fetched data and edits
+ */
+export function getCurrentlyEditingProduct( state ) {
```

Comment:
Ah - the future quick UI especially makes sense. As for the `creates` - wouldn't there only ever be zero or one products in there at any time?